src/pt/argoscomics/src/eu/kanade/tachiyomi/extension/pt/argoscomics/ArgosComics.kt

```diff
@@ -1,63 +1,250 @@
+        val latestHeaders = headers.newBuilder()
+            .set("Next-Action", latestToken)
+            .build()
+
+        val payload = emptyList<String>()
+            .toJsonString()
+            .toRequestBody(TEXT_PLAIN_MEDIA_TYPE)
+
+        return POST(baseUrl, latestHeaders, payload)
+    }
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val dto = getJsonBody(response).parseAs<List<MangaDto>>()
+        return MangasPage(dto.map(MangaDto::toSManga), false)
+    }
+
+    // ======================== Search =============================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$baseUrl/projetos".toHttpUrl().newBuilder()
+            .addQueryParameter("page", page.toString())
+            .build()
+
+        val payload = listOf(query)
+            .toJsonString()
+            .toRequestBody(TEXT_PLAIN_MEDIA_TYPE)
 
-    override val useNewChapterEndpoint = true
+        val searchHeaders = headers.newBuilder()
+            .set("Next-Action", searchToken)
+            .build()
+
+        return POST(url.toString(), searchHeaders, payload)
+    }
+
+    override fun searchMangaParse(response: Response) = latestUpdatesParse(response)
+
+    // ======================== Details =============================
+
+    override fun getMangaUrl(manga: SManga) = "$baseUrl/${manga.url}"
+
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        val payload = manga.url.split("/").filter(String::isNotBlank)
+            .toJsonString()
+            .toRequestBody(TEXT_PLAIN_MEDIA_TYPE)
+
+        val detailsHeaders = headers.newBuilder()
+            .set("Next-Action", getDetailsToken(getMangaUrl(manga)))
+            .build()
+
+        return POST("$baseUrl${manga.url}", detailsHeaders, payload)
+    }
+
+    override fun mangaDetailsParse(response: Response) =
+        getJsonBody(response).parseAs<MangaDetailsDto>().toSManga()
+
+    // ======================== Chapters =============================
+
+    override fun chapterListRequest(manga: SManga): Request {
+        val chaptersHeaders = headers.newBuilder()
+            .set("Next-Action", getChaptersToken(getMangaUrl(manga)))
+            .build()
+
+        return mangaDetailsRequest(manga).newBuilder()
+            .headers(chaptersHeaders)
+            .build()
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val pathSegment = response.request.url.toString().substringAfter(baseUrl)
+        return getJsonBody(response).parseAs<VolumeChapterDto>().toChapterList(pathSegment)
+    }
+
+    // ======================== Pages =============================
+
+    override fun getChapterUrl(chapter: SChapter): String = "$baseUrl${chapter.url}"
+
+    override fun pageListRequest(chapter: SChapter): Request {
+        val segments = chapter.url.split("/").filter(String::isNotBlank)
```

Comment:
If you're working on segments already, you should prefer using `HttpUrl` for this, to avoid matching stuff in the query arguments / fragment. Not tested:

```suggestion
        val segments = "$baseUrl${chapter.url}".toHttpUrl()!!.pathSegments
```